artcommon/artcommonlib/util.py

```
@@ -382,59 +384,60 @@ def get_inflight(assembly, group, date=None):
+                        best_date = release_date
+                else:
+                    LOGGER.warning(f"Release '{release['name']}' qualified but has no X.Y.Z version in its name")
 
-        if not inflight_release:
-            LOGGER.info(
-                f'Did not find a {prev_group} release that is releasing ~ in the same week as {assembly} {assembly_release_date}'
+        if best_candidate:
+            inflight_release = best_candidate
+
+    except json.JSONDecodeError as e:
+        raise ValueError(f'Failed to parse JSON for {prev_group}: {e}')
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)